src/rdkafka_telemetry_encode.c

```diff
@@ -0,0 +1,869 @@
+                            rkb->rkb_avg_outbuf_latency.ra_v.sum;
+                        int64_t historic_sum =
+                            rkb->rkb_c_historic.rkb_avg_outbuf_latency.ra_v.sum;
+                        int64_t cnt_diff = current_cnt - historic_cnt;
+                        int64_t sum_diff = current_sum - historic_sum;
+
+                        sum_value +=
+                            sum_diff /
+                            (cnt_diff * RDKAFKA_TELEMETRY_NS_TO_MS_FACTOR);
+                }
+        }
+        avg_queue_time.intValue = sum_value / broker_count;
+        return avg_queue_time;
+}
+
+static rd_kafka_telemetry_metric_value_t
+calculate_queue_time_max(rd_kafka_t *rk, rd_kafka_broker_t *broker) {
+        rd_kafka_telemetry_metric_value_t max_queue_time;
+        rd_kafka_broker_t *rkb;
+
+        max_queue_time.intValue = 0;
+        TAILQ_FOREACH(rkb, &rk->rk_brokers, rkb_link) {
+                max_queue_time.intValue =
+                    RD_MAX(max_queue_time.intValue,
+                           rkb->rkb_avg_outbuf_latency.ra_v.maxv_interval);
+        }
+        max_queue_time.intValue /= RDKAFKA_TELEMETRY_NS_TO_MS_FACTOR;
+        return max_queue_time;
+}
+
+static rd_kafka_telemetry_metric_value_t
+calculate_consumer_assigned_partitions(rd_kafka_t *rk,
+                                       rd_kafka_broker_t *broker) {
+        rd_kafka_telemetry_metric_value_t assigned_partitions;
+        rd_kafka_broker_t *rkb;
+        int32_t total_assigned_partitions = 0;
+
+        TAILQ_FOREACH(rkb, &rk->rk_brokers, rkb_link) {
+                total_assigned_partitions += rkb->rkb_toppar_cnt;
+                total_assigned_partitions -=
+                    rkb->rkb_c_historic.assigned_partitions;
+        }
+        assigned_partitions.intValue = total_assigned_partitions;
+        return assigned_partitions;
+}
+
+
+static void reset_historical_metrics(rd_kafka_t *rk) {
+        rd_kafka_broker_t *rkb;
+
+        TAILQ_FOREACH(rkb, &rk->rk_brokers, rkb_link) {
+                rkb->rkb_c_historic.assigned_partitions = rkb->rkb_toppar_cnt;
+                rkb->rkb_c_historic.connects = rkb->rkb_c.connects.val;
+                rkb->rkb_c_historic.ts_last  = rd_uclock() * 1000;
+                rkb->rkb_c_historic.connects = rkb->rkb_c.connects.val;
+                /* Only ra_v is being used to keep track of the metrics */
+                rkb->rkb_c_historic.rkb_avg_rtt.ra_v = rkb->rkb_avg_rtt.ra_v;
+                rd_atomic32_set(&rkb->rkb_avg_rtt.ra_v.maxv_reset, 1);
+                rkb->rkb_c_historic.rkb_avg_throttle.ra_v =
+                    rkb->rkb_avg_throttle.ra_v;
+                rd_atomic32_set(&rkb->rkb_avg_throttle.ra_v.maxv_reset, 1);
+                rkb->rkb_c_historic.rkb_avg_outbuf_latency.ra_v =
+                    rkb->rkb_avg_outbuf_latency.ra_v;
+                rd_atomic32_set(&rkb->rkb_avg_outbuf_latency.ra_v.maxv_reset,
+                                1);
+        }
```

Comment:
Here it just needs to set the `ts_last` in rk and the historic connect count.
```suggestion

        rk->rk_telemetry.rk_historic_c.ts_last = now_ns;
        TAILQ_FOREACH(rkb, &rk->rk_brokers, rkb_link) {
                rkb->rkb_telemetry.rkb_historic_c.connects =
                    rd_atomic32_get(&rkb->rkb_c.connects);
        }
```

Reply:
Updated

Reply:
This isn't needed. Removed now